### ISSUE TITLE
HVF QEMU fails on Intel - path to qemu is invalid

### ISSUE THREAD

**Issue by darkn3rd** [ppggff/vagrant-qemu#59]
I wanted to try out this plugin on an Macbook (Intel), but it seems to crash/burn.

### STEPS

```bash
## Install
brew install qemu
brew tap hashicorp/tap
brew install hashicorp/tap/hashicorp-vagrant
vagrant plugin install vagrant-qemu

cat << EOF > Vagrantfile
Vagrant.configure("2") do |config|
  config.vm.box = "generic/ubuntu2204"
  config.vm.provider "qemu" do |qe|
    qe.ssh_port = "50022" # change ssh port as needed
  end
end
EOF

vagrant up --provider=qemu
```

## Expected Results

I expected that the system would come up, as the `generic/ubuntu2204` image supports QEMU. 

## Actual Results

Something has hardwired the Homebrew path for ARM64 binaries of ` /opt/homebrew/`, and not the Intel_x86_64 binaries, which is `/usr/local`.  These can be resolved by using the $HOMEBREW_PREFIX.

```
Invalid config.

Error: Invalid qemu dir: /opt/homebrew/share/qemu
```

**Comment by ppggff** [comment]
Please try the x86_64 example in the Readme.

**Comment by darkn3rd** [comment]
I will test these out.  Why would one not want to use `hvf` as it is supported by QEMU on Intel.

**Comment by darkn3rd** [comment]
### Example 4 Test: FAILS as documented

1. Docs unclear or ambiguous if this is an Intel example for Macs on Apple Silicon, Macs on Intel, or both. 
2. `centos/7` does not have `qemu` listed as a supported provider, so `vagrant up --provider=qemu` uses `libvirt` provider.
3. This test failed, but the plugin defaults the wrong path to `qemu` binary.  I expected that (1) homebrew environment would be detected, (2) given homebrew environment search for `HOMEBREW_PREFIX` env var, (3) if `HOMEBREW_PREFIX` env var not found select path based on current arch, `/usr/local/share/qemu` for Mac Intel host, `/opt/homebrew/share/qemu` for Mac on Apple Silicon host. 

### Steps
```bash
cat << EOF > Vagrantfile 
# Example 4 form docs
Vagrant.configure(2) do |config|
  config.vm.box = "centos/7"

  config.vm.provider "qemu" do |qe|
    qe.arch = "x86_64"
    qe.machine = "q35"
    qe.cpu = "qemu64"
    qe.net_device = "virtio-net-pci"
  end
end
EOF  

vagrant up --provider=qemu
```

### Actual Results

Example 4 as documented does fails on Intel Macs.

```
Invalid config.

Error: Invalid qemu dir: /opt/homebrew/share/qemu
```

**Comment by darkn3rd** [comment]
TL;DR summary is that there should be an intelligent default solution path for macOS (Intel) to use HVF.  

For the examples, separate from this issue, there should be clarity as to the host the example is appropriate for.  Right it is vague and ambiguous, especially for those new to QEMU.  As a suggestion, perhaps have examples that or organized per host.  There may be some redundancy.  Or if that is not welcomed, maybe having some text to show the type of host.

**Comment by ppggff** [comment]
Thanks, I will try to make it more intelligent.

**Comment by compte** [comment]
QEMU fails for similar reasons on Windows and Linux.
I'm using module OS to detect the system.
If @ppggff can integrate this in the plugin, it would be great and make our life easier. But you can also update an example if it's easier for you.

      if OS.windows?
      then
        # usually similar to qe.qemu_dir = "C:\\Program Files\\qemu\\share"
        ProgramFiles = ENV["ProgramFiles"]
        qe.qemu_dir = "#{ProgramFiles}\\qemu\\share"
      else
        if OS.linux?
        then
          qe.qemu_dir = "/usr/share/qemu"
        else
          # tuning for MacOS goes here, educated guess
          HomebrewPrefix= Env["HOMEBREW_PREFIX"]||"/usr/local" 
          qe.qemu_dir = "#{HomebrewPrefix}/share/qemu"
        end
      end
      # define an environment variable QEMU_DIR if the above logic fails and restart.
      qe.qemu_dir = ENV["QEMU_DIR"] || qe.qemu_dir

      #same logic (host OS detection) could be used for other stuff like qe.no_daemonize = OS.windows?

**Comment by ppggff** [comment]
This should be fixed in **v0.4.0** (now on RubyGems).

The provider is now host-aware: on an Intel Mac it defaults to `q35` + `accel=hvf` with `cpu=host` (and falls back to `accel=tcg` + `cpu=max` when emulating a different guest arch), resolves `qemu_dir` from `QEMU_DIR` / `HOMEBREW_PREFIX`, and skips the `qemu_dir` check entirely for x86_64 (SeaBIOS) — so a plain Intel-Mac config no longer fails with "Invalid qemu dir". MacPorts (or any non-Homebrew layout) is covered by setting `QEMU_DIR`.

```sh
vagrant plugin install vagrant-qemu   # or: vagrant plugin update vagrant-qemu
```

Could you give it a try on your Intel Mac and let me know if it works? Thanks!
